app/scripts/modules/amazon/src/loadBalancer/details/loadBalancerDetails.controller.ts

```diff
@@ -212,6 +212,9 @@ export class AwsLoadBalancerDetailsController {
   public getFirstSubnetPurpose(subnetDetailsList: ISubnet[] = []) {
     return head(subnetDetailsList.map(subnet => subnet.purpose)) || '';
   };
+
+  // Satisfy TypeScript 2.4 breaking change: https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#weak-type-detection
+  public $onInit() {}
```

Comment:
I tried to update @types/angular... They fixed the generics bit, but broke angular-mocks, which I would be super happy if you could figure that out.